internal/bminventory/inventory.go

```diff
@@ -1881,7 +1887,16 @@ func (b *bareMetalInventory) validateAndUpdateClusterParams(ctx context.Context,
 	log := logutil.FromContext(ctx, b.log)
 
 	if swag.StringValue(params.ClusterUpdateParams.PullSecret) != "" {
-		if err := b.ValidatePullSecret(*params.ClusterUpdateParams.PullSecret, ocm.UserNameFromContext(ctx), cluster.OcpReleaseImage); err != nil {
+		var mirroredRegistries []string
+		if cluster.MirrorRegistryConfiguration != "" {
+			mirrorRegistriesConfig, err := cluster.GetMirrorRegistryConfiguration()
+			if err != nil {
+				b.log.WithError(err).Warnf("failed to get saved cluster mirror registry configuration while updating cluster params")
+			} else {
+				mirroredRegistries = extractMirroredRegistriesFromConfig(log, mirrorRegistriesConfig)
```

Comment:
What happens if extraction fails?

Reply:
As agreed below let's put the comment on extractMirroredRegistriesFromConfig, that way we should only need to do it once.
I think it's reasonable for the reader to look at the function and see the comment.

Reply:
Logging within the function makes sense.

There is an interesting edge case here though...

We are within a block of code where we expect that the mirror registry configuration has been set.
So if NO registries come back from this call I could argue that this is unexpected behaviour.

We later rely on this result to decide on the registries that will be ignored, I am not sure whether a problem here could have an unintended side effect?

Perhaps we want to consider this case to be a full error?

Reply:
Thanks for the review and discussion, Paul!

Discussed with Paul offline:

- If there's a failure with parsing mirror registries here, then it should not fail at this point in the function. 
    - The failure to parse mirror registries will also fail the validation of the pull secret. 
    - So the failure will show up to the user.
- Also if there's a failure to parse the mirror registries, it should also fail elsewhere. 
    - Want to verify with @eliorerz that this is the case too.

Will also add the ignored registries passed in to the logging to help debug.

Reply:
Thanks Paul! I've also added the ignored registries in the error message during pull secret validation so we can more easily see what registries were passed in.